### PR TITLE
Refactor ROS 2 System components

### LINKED ISSUE #626: Refactor SystemComponent to remove dependency on RPI
**Is your feature request related to a problem? Please describe.**

Users will want to run headless simulations with ROS 2 Gem. Currently SystemComponent includes Camera pass system initialization, which creates an unwanted dependency. 

**Describe the solution you'd like**
Camera system component should be created and handle well the case of headless (no pass, camera sensors should be inactive).

**Describe alternatives you've considered**
No good alternatives, it is good for encapsulation

**Additional context**
Running headless simulation is very useful for scenario-based validation.

### PR DESCRIPTION
**Pull request by michalpelka**
Fixes #626

### CODE REVIEW

File: Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp:1]: Directory "SystemComponents" could be misleading, as it has only some of system components, but not all of them.
Perhaps the new camera SC should go to the Camera folder instead (much like it is with lidar)?

File: Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+           [this]()
+           {
+               this->LoadPassTemplateMappings();
+           });
+       passSystem->ConnectEvent(m_loadTemplatesHandler);
+   }
+
+   void ROS2SystemCameraComponent::Activate()
+   {
+
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp:70]:
```suggestion
```

File: Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+   {
+       required.push_back(AZ_CRC_CE("ROS2Service"));
+       required.push_back(AZ_CRC_CE("RPISystem"));
+   }
+
+   void ROS2SystemCameraComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+   {
+   }
+
+   void ROS2SystemCameraComponent::Init()
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp:51]: Empty functions can be removed

File: Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+{
+   void ROS2SystemCameraComponent::Reflect(AZ::ReflectContext* context)
+   {
+       if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+       {
+           serialize->Class<ROS2SystemCameraComponent, AZ::Component>()->Version(0);
+
+           if (AZ::EditContext* ec = serialize->GetEditContext())
+           {
+               ec->Class<ROS2SystemCameraComponent>("ROS2 System Camera Component", "This system component is responsible for the ROS2 Camera simulation.")
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp:22]:
```suggestion
               ec->Class<ROS2SystemCameraComponent>("ROS 2 System Camera Component", "This system component is responsible for setting a pass template for simulation of camera sensors.")
```

File: Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+   {
+       AZ_Printf("ROS2SystemCameraComponent", "LoadPassTemplateMappings\n");
+       auto* passSystem = AZ::RPI::PassSystemInterface::Get();
+       AZ_Assert(passSystem, "PassSystemInterface is null");
+
+       const char* passTemplatesFile = "Passes/ROSPassTemplates.azasset";
+       [[maybe_unused]] bool isOk = passSystem->LoadPassTemplateMappings(passTemplatesFile);
+       AZ_Assert(isOk, "LoadPassTemplateMappings return false ");
+   }
+
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemComponent.cpp:94]:
```suggestion
```

File: Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp [hunk 6]
```diff
@@ -30,6 +30,7 @@ namespace ROS2
         ROS2SpawnerComponentBase::Activate();
 
         auto ros2Node = ROS2Interface::Get()->GetNode();
+        AZ_Assert(ros2Node, "ROS2 node is not initialized");
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp:33]:
```suggestion
        AZ_Assert(ros2Node, "ROS 2 node is not initialized");
```

Comment (adamdbrw) [reply Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp:33]: This should be for the whole codebase.

File: Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemEditorComponent.cpp [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+    void ROS2EditorCameraSystemComponent::OnStartPlayInEditorBegin()
+    {
+        BaseSystemComponent::Activate();
+    }
+    void ROS2EditorCameraSystemComponent::OnStopPlayInEditor()
+    {
+        BaseSystemComponent::Deactivate();
+    }
+
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemEditorComponent.cpp:68]:
```suggestion
```

File: Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemEditorComponent.h [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+ */
+#pragma once
+
+#include "ROS2CameraSystemComponent.h"
+
+#include <AzToolsFramework/Entity/EditorEntityContextBus.h>
+
+namespace ROS2
+{
+    /// System component for ROS2 editor
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/SystemComponents/ROS2CameraSystemEditorComponent.h:16]: Update documentation

**Comment by michalpelka** [comment]
The services that are provided in system components are not global, they are available only in System Entity.

**Review by jhanca-robotecai — CHANGES_REQUESTED** [review]
The code builds and runs correctly.

Please fix the duplicated header reference in cmake file. The remaining comments can be considered nitpicks.

File: Gems/ROS2/Code/Source/Camera/ROS2CameraSystemEditorComponent.h
[the remark this answers is not in the document]
Comment (michalpelka) [reply Gems/ROS2/Code/Source/Camera/ROS2CameraSystemEditorComponent.h:1]: Renamed ROS2EditorCameraSystemComponent.

**Comment by michalpelka** [comment]
@jhanca-robotecai updated my contribution.

File: Gems/ROS2/Code/Source/Camera/ROS2CameraSystemComponent.h [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+       AZ_COMPONENT(ROS2SystemCameraComponent, "{b4665d39-78fd-40de-8518-2f6bd345a831}");
+
+       static void Reflect(AZ::ReflectContext* context);
+
+       static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+       static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+       static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+
+       ROS2SystemCameraComponent() = default;
+       ~ROS2SystemCameraComponent() override = default;
```
[the remark this answers is not in the document]
Comment (michalpelka) [reply Gems/ROS2/Code/Source/Camera/ROS2CameraSystemComponent.h]: Technically, we override the virtual destructor in  AZ::Component.

**Review by jhanca-robotecai — APPROVED** [review]
All my remarks were taken into consideration, no more issues from my side.

**Pull request merged**